design/open/openapi-multipart-upload.md

```diff
@@ -0,0 +1,227 @@
+      required:
+        - upload_id
+        
+    UploadPart:
+      type: object
+      properties:
+        part_number:
+          type: integer
+        etag:
+          type: string
+      required:
+        - part_number
+        - etag
+        
+    CompleteMultipartUpload:
+      type: object
+      properties:
+        parts:
+          type: array
+          description: "List of uploaded parts, should be ordered by ascending part number"
+          items:
+            $ref: "#/components/schemas/UploadPart"
+      required:
+        - parts
+```
+
+### Support and discover
+
+Presign support is a capability lakectl discover before switching to use presign for upload or download from lakeFS.
+The multipart upload support will be part of the storage capability add `multipart_upload_support` optional field that when set to `true` the user can perform multipart upload using the new API.
+The client will need to check also for `pre_sign_support` in order to request multipart upload with `presigned_parts`.
+Currently as we only support multipart upload for persigned configuration, the server will turn off multipart upload support when presigned is false.
+
+```yaml
+    StorageConfig:
+      type: object
+      required:
+        - blockstore_type
+        - blockstore_namespace_example
+        - blockstore_namespace_ValidityRegex
+        - pre_sign_support
+        - pre_sign_support_ui
+        - import_support
+        - import_validity_regex
+      properties:
+        blockstore_type:
+          type: string
+        blockstore_namespace_example:
+          type: string
+        blockstore_namespace_ValidityRegex:
+          type: string
+        default_namespace_prefix:
+          type: string
+        pre_sign_support:
+          type: boolean
+        pre_sign_support_ui:
+          type: boolean
+        import_support:
+          type: boolean
+        import_validity_regex:
+          type: string
+        pre_sign_support_parts:
+          type: boolean
+```
+
+## Limitations
+
+- **S3 block adapter exclusive:** Uploading files in multiple parts (multipart upload) is only available when using the S3 block adapter. This feature isn't currently supported with other storage options. This exclude tha multipart upload support we provide by our S3 gateway where we provide implementation above each storage.
+- **Part size and count restrictions:** There are limits on how you can split your file for upload:
+    - **Maximum parts:** You can split your file into a maximum of 10000 parts.
+    - **Minimum part size:** Each part must be at least 5MB in size. This is a temporary constraint and isn't currently configurable or discoverable. It will become an option when additional storage options are supported.
+- **Initiating the upload:** When starting a multipart upload, you'll need to specify the total number of parts in your request. It reduce the requests for each part presigned URL when the client already knows the size.
```

Comment:
```suggestion
- **Initiating the upload:** When starting a multipart upload, you'll need to specify the total number of parts in your request. It will reduce the requests for each part's presigned URL when the client already knows the size.
```